packages/app/src/app/components/CodeEditor/Monaco/index.tsx

```diff
@@ -430,85 +489,105 @@ class MonacoEditor extends React.Component<Props, State> implements Editor {
         // that the changes of code are not sent to live users. We need to reset
         // this state when we're doing changing modules
         this.props.onCodeReceived();
-        this.liveOperationCode = '';
       }
     });
   };
+  /**
+   * Throttle the changes and handle them after a desired amount of time as one array of changes
+   */
+  addChangesOperation = (changes: any[]) => {
+    // Module changed in the meantime
+    if (
+      this.changes.moduleShortid &&
+      this.changes.moduleShortid !== this.currentModule.shortid
+    ) {
+      this.sendChangeOperations();
+    }
 
-  onSelectionChangedDebounced = data => {
-    if (this.props.onSelectionChanged) {
-      this.props.onSelectionChanged(data);
+    if (!this.changes.code) {
+      this.changes.code = this.currentModule.code || '';
+    }
+    if (!this.changes.moduleShortid) {
+      this.changes.moduleShortid = this.currentModule.shortid;
     }
+
+    changes.forEach(change => {
+      this.changes.changes.push(change);
+    });
+
+    this.sendChangeOperations();
   };
 
-  liveOperationCode = '';
-  sendChangeOperations = changeEvent => {
+  sendChangeOperations = (retry: boolean = false) => {
```

Comment:
We should have the other code here, so `sendChangeOperations = changeEvent` and `liveOperationCode = ''` etc.